third_party/move/tools/move-resource-viewer/src/fat_type.rs

```diff
@@ -50,7 +50,24 @@ pub(crate) struct FatStructType {
     pub name: Identifier,
     pub abilities: WrappedAbilitySet,
     pub ty_args: Vec<FatType>,
-    pub layout: Vec<FatType>,
+    pub layout: FatStructLayout,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) enum FatStructLayout {
+    Singleton(Vec<FatType>),
+    Variants(Vec<Vec<FatType>>),
```

Comment:
Probably better to move the complexity out (both variants are vec<X>) and seems like the accesses to them are also monotone

Reply:
Not convinced. I think a vec-of-vec is a natural representation for the core type info.

Also note this thing is obsolete and should be replaced, so not too much excitement about it.